src/main/java/duke/task/TaskArray.java

```diff
@@ -0,0 +1,367 @@
+                updateFile(current_task.toString(), keyword);
+            }
+
+
+        } catch (NumberFormatException n) {
+            //Alerts user upon not entering a task number
+            PrintMethod.printInvalidDone();
+
+        } catch (IllegalNumberException i) {
+            //Alerts user upon entering a task that has not been initialised
+            PrintMethod.printInvalidTask();
+
+        } catch (IndexOutOfBoundsException o) {
+            //Alerts user upon entering a number that is out of range
+            PrintMethod.printWithinRangeTask();
+
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+        PrintMethod.printLines();
+
+    }
+
+    public static int findTaskNumber(String command) throws IllegalNumberException {
+
+        int indexOfTask;
+        indexOfTask = Integer.parseInt(command);
+        if (indexOfTask > taskArrayList.size() && indexOfTask <= MAX_SIZE) {
+            throw new IllegalNumberException();
+        }
+        return indexOfTask - 1;
+
+    }
+
+
+    public static void addToTasks(Task task, boolean fileWrite) throws IOException {
+        taskArrayList.add(task);
+        task.addTask();
+        if (fileWrite) {
+            writeToFile(task.toString());
+            System.out.println("Now you have " + taskArrayList.size() + " tasks in the list.");
+        }
+    }
+
+
+    public static void printTasks() {
+        PrintMethod.printAllTasks(taskArrayList);
+    }
+
+    /**
+     * Returns the preposition and the deadline of the given task.
+     *
+     * @param typeOfTask The type of the task in the command.
+     * @param command    The command given by the user.
+     * @return The task and the deadline for the task as a String array
+     * @throws IllegalEmptyDescriptionException If the task is not given by the user
+     * @throws IllegalPrepositionWithoutDate    If the user does not enter the date after the preposition
+     */
+    public static String[] parseString(String typeOfTask, String command) throws IllegalEmptyDescriptionException,
+            IllegalPrepositionWithoutDate {
+
+        String[] words = command.split(DELIMITER_SLASH);
+        String task = returnTaskDescription(words[0], typeOfTask);
+        words[0] = task;
+        String deadlineForTask = words[1].strip();
+
+
+        //splits the deadline into its preposition and the date
+        String[] descriptorsForDate = deadlineForTask.split(DELIMITER_CHARACTER);
+        String preposition = descriptorsForDate[0].strip();
+        deadlineForTask = deadlineForTask.replace(preposition, DELIMITER_EMPTY_STRING).strip();
+
+        if (deadlineForTask.equals(DELIMITER_EMPTY_STRING)) {
+            throw new IllegalPrepositionWithoutDate();
+        }
+
+        words[1] = preposition + DELIMITER_SEMI_COLON + deadlineForTask;
+        return words;
+
+    }
+
+    public static void fileData(ArrayList<String> filesLines) {
```

Comment:
It will be more readable if the name of the method is an action.

Reply:
Think again, maybe what you mean here is the file is the verb. Please ignore this comment.